modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/behavior/TextInputControlBehavior.java

```diff
@@ -620,18 +623,7 @@ private void rightWord() {
     }
 
     protected void fire(KeyEvent event) { } // TODO move to TextFieldBehavior
-    protected void cancelEdit(KeyEvent event) { forwardToParent(event);} // not autoconsumed
-
-    protected void forwardToParent(KeyEvent event) {
-        // fix for JDK-8145515
```

Comment:
Will this code removal re-introduce JDK-8145515?

Reply:
no don't think so: the fix for JDK-8145515 was to prevent the triggering of parent.fireEvent, so removing the method altogether should have no effect :) Could add a test to explicitly guard against this.

Please note that the old combo issue is back in the form of https://bugs.openjdk.java.net/browse/JDK-8229914 - before and after fixing this, combo does very bad things ;)

Reply:
just to let you know: locally added test to explicitly guard against JDK-8145515 (and an ignored one for the regression JDK-8229914) - they are passing/failing the same before/after this fix. Will push later ...

Reply:
pushed

Reply:
fwiw: wrote a short summary of enter handling in comboBox (that is JDK-8145515 and related) https://github.com/kleopatra/swingempire-fx/wiki/ComboBoxEnterHandling